packages/@aws-cdk/aws-rds/test/cluster.test.ts

```diff
@@ -762,9 +822,101 @@ describe('cluster', () => {
     cluster.addRotationSingleUser();
 
     // THEN
-    expect(() => cluster.addRotationSingleUser()).toThrow(/A single user rotation was already added to this cluster/);
+    expect(() => cluster.addRotationSingleUser()).toThrow(/A single user rotation for master secret was already added to this cluster/);
+  });
+
+  test('throws when trying to add single user rotation for additional secret which is master secret', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+
+    // WHEN
+    cluster.addRotationSingleUser();
+
+    // THEN
+    expect(() => {
+      cluster.addRotationSingleUser({ secret: cluster.secret });
+    }).toThrow(/A single user rotation for master secret was already added to this cluster/);
+  });
+
+  test('add single user rotation for two additional secrets', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+    const secret1 = new DatabaseSecret(stack, 'MySecret1', { username: 'user1' });
+    secret1.attach(cluster);
+    const secret2 = new DatabaseSecret(stack, 'MySecret2', { username: 'user2' });
+    secret2.attach(cluster);
+
+    // WHEN
+    cluster.addRotationSingleUser({ secret: secret1 });
+    cluster.addRotationSingleUser({ secret: secret2 });
+
+    // THEN
+    expect(stack).toHaveResource('AWS::SecretsManager::RotationSchedule', {
+      SecretId: {
+        Ref: 'MySecret1E18D875B',
+      },
+      RotationLambdaARN: {
+        'Fn::GetAtt': [
+          'MySecret1RotationSingleUser7D84B21E',
+          'Outputs.RotationLambdaARN',
+        ],
+      },
+    });
+    expect(stack).toHaveResource('AWS::SecretsManager::RotationSchedule', {
+      SecretId: {
+        Ref: 'MySecret225E18D70',
+      },
+      RotationLambdaARN: {
+        'Fn::GetAtt': [
+          'MySecret2RotationSingleUser2903C638',
```

Comment:
Hmm. This is interesting. Are we adding a different Lambda per each rotation? Is that a good idea? Perhaps these should share the same Lambda? I guess before, this was not a concern, because you could only have a single one-user rotation (with `addSingleUserRotation()`), but now that you can have multiples, I think we should not be creating a separate Lambda for each of them.

Reply:
No strong opinion. But I'm not sure how to implement it thou. I'm using the `singleUserRotationApplication` of the (cluster) instance.

Reply:
BTW, if we go with single-lamba approach, it is gonna be inconsistent with multi-user rotation feature.